src/main/java/org/tron/common/runtime/vm/program/Stack.java

```diff
@@ -59,4 +60,25 @@ public void swap(int from, int to) {
   private boolean isAccessible(int from) {
     return from >= 0 && from < size();
   }
+
+  @Override
+  public boolean equals(Object o) {
+    if (this == o) {
+      return true;
+    }
+    if (!(o instanceof Stack)) {
```

Comment:
![Bug](https://sonarsource.github.io/sonar-github/bug.png 'Bug') Bug: Compare to "this.getClass()" instead. ([squid:S2162](https://sonarcloud.io/organizations/tron-zhaohong/rules#rule_key=squid%3AS2162))

[See it in SonarCloud](https://sonarcloud.io/project/issues?id=java-tron&pullRequest=1681&issues=AWbnEHQ6SgN4uCcljsr2&open=AWbnEHQ6SgN4uCcljsr2)

[ISSUE_KEY]: AWbnEHQ6SgN4uCcljsr2